infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -142,9 +326,149 @@ private boolean registryTableHasNewEntries() {
+                    List<CorfuStoreEntry<ClientRegistrationId, ClientRegistrationInfo, Message>> registrationResults =
+                            txn.executeQuery(clientRegistrationTable, record -> true);
+                    registrationResults.forEach(entry -> {
+                        String clientName = entry.getKey().getClientName();
+                        ReplicationModel model = entry.getPayload().getModel();
+                        ReplicationSubscriber subscriber = ReplicationSubscriber.newBuilder()
+                                .setClientName(clientName).setModel(model).build();
+                        // TODO (V2): currently we don't support ccustomized client name, default logical group subscriber
+                        //  should be removed after the grpc stream for Sink session creation is created.
+                        if (model.equals(ReplicationModel.LOGICAL_GROUPS)) {
+                            subscriber = getDefaultLogicalGroupSubscriber();
+                        }
+                        registeredSubscribers.add(subscriber);
+                    });
+
+                    List<CorfuStoreEntry<ClientDestinationInfoKey, DestinationInfoVal, Message>> groupConfigResults =
+                            txn.executeQuery(clientConfigTable, record -> true);
+                    groupConfigResults.forEach(entry -> {
+                        String clientName = entry.getKey().getClientName();
+                        ReplicationModel model = entry.getKey().getModel();
+                        ReplicationSubscriber subscriber = ReplicationSubscriber.newBuilder()
+                                .setClientName(clientName).setModel(model).build();
+                        String groupName = entry.getKey().getGroupName();
+                        // TODO (V2): currently we don't support ccustomized client name, default logical group subscriber
+                        //  should be removed after the grpc stream for Sink session creation is created.
+                        if (model.equals(ReplicationModel.LOGICAL_GROUPS)) {
+                            subscriber = getDefaultLogicalGroupSubscriber();
+                        }
+                        if (registeredSubscribers.contains(subscriber)) {
+                            groupSinksMap.put(groupName, new HashSet<>(entry.getPayload().getDestinationIdsList()));
+                        } else {
+                            log.warn("Subscriber {} not registered, but found its group config in client table: {}",
+                                    subscriber, groupName);
+                        }
+                    });
+
+                    return txn.commit();
+                } catch (TransactionAbortedException tae) {
+                    log.error("Failed to preprocess client configuration tables due to Txn Abort", tae);
+                    throw new RetryNeededException();
+                }
```

Comment:
what would be the return value here if retry has been exhausted?

Reply:
I think if retry exhausted the exception will be thrown and executing thread will be terminated, which is expected as we shouldn't proceed without the client config listener running.